courses/machine_learning/deepdive2/feature_engineering/solutions/3_Basic_Feature_Engineering_in_Keras.ipynb

```diff
@@ -0,0 +1,2501 @@
+{
```

Comment:
<p>Good job checking for the gpu version!</p><p> </p><p>Something must have changed with the notebook environments because it will now throw a permission error and will suggest adding the --user flag, so something like:</p><p> </p><p>!pip install --user sklearn</p><p>!python3 -m pip freeze | grep 'tensorflow==2\|tensorflow-gpu==2' || \</p><p>&nbsp;&nbsp;python3 -m pip install --user tensorflow==2</p><p> </p><p>?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/777/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Reply:
This is a weird one. I experimented with a new instance and did receive the same permissions error for sklearn.  I added --user (e.g. !pip install --user sklearn) and it ran but still gave an error.  However, when I opened up a new notebook and installed sklearn, it showed that the requirements were already satisfied.  I'll restart the kernel after I run through the lab to ensure that the installs run smoothly the first time.